common/lib/dataset.py

```diff
@@ -517,11 +538,13 @@ def delete(self, commit=True):
 			try:
 				child = DataSet(key=child["key"], db=self.db)
 				child.delete(commit=commit)
+				child.delete_annotations()
```

Comment:
This is redundant since `child.delete()` will call the same `delete_annotations()` you added below (it's recursive)